guides/source/form_helpers.md

```diff
@@ -921,14 +1052,16 @@ Which will result in a `params` hash that looks like:
 
 All of the form inputs map to the `"person"` hash because we called `fields_for`
 on the `person_form` form builder. Also, by specifying `index: address.id`, we
-rendered the `name` attribute of each city input as `person[address][#{address.id}][city]`
-instead of `person[address][city]`. Thus we are able to determine which Address
-records should be modified when processing the `params` hash.
+rendered the `name` attribute of each city input as
+`person[address][#{address.id}][city]` instead of `person[address][city]`. This
```

Comment:
I think this whole section should use `addresses` instead of `address` for the form/parameters.
The association on person would probably be `addresses` as well and other examples above use `addresses` as well (this suggestion doesn't include all occurences as that was a bit diffuclt to do in Github 😞 )
```suggestion
`person[addresses][#{address.id}][city]` instead of `person[addresses][city]`. This
```

Hmm, the next section implements this very similar example with nested attributes. Maybe this section should use a single address for a person instead?

Reply:
Hi @p8 - let me know if I misunderstood something but I double checked and it is indeed `address` as it is coming from within the loop @person.addresses.each do |address|:

```ruby
  <% @person.addresses.each do |address| %>
    <%= person_form.fields_for address, index: address.id do |address_form| %>
      <%= address_form.text_field :city %>
    <% end %>
  <% end %>
```
The `id` specified using the `index:` option is what will then distinguish the addresses with `person[address][23][city]`, I think that's the "feature" this section is trying to illustrate. Let me know if that make sense, or if I've misunderstood what you're getting it happy to look into it more.